### DISCUSSION TITLE
DBRider integration with AgroalDataSource

### DISCUSSION THREAD

**Discussion by nicklasweasel** [quarkusio/quarkus#41027] [Community]
```
@QuarkusTest
@DBRider
class TestDao {
    @Inject
    Dao dao;

    @Inject
    AgroalDataSource agroalDataSource;

    ConnectionHolder connectionHolder = () -> agroalDataSource.getConnection();

    @Test
    @DataSet("products/delete_product_pre.json")
    @ExpectedDataSet("products/delete_product_post.json")
    void testDeleteProduct() {
        dao.deleteProduct(1);
    }
}
```

Anyone managed to integrate DBRider with @QuarkusTest?

if I comment out @DBRider, the test is executed fine but with the integration enabled, I get an error about "this.agroalDataSource" is null". If I move to constructor injection and place a breakpoint there, I can actually see that the constructor is called twice, the first time with a null agroalDataSource and the second time with a valid one.

Another odd thing is that if I make the connectionHolder lambda breakpointable, I never see it hit before it goes kaboom.

Any pointers or experiences?

**Comment by edeandrea** [comment]
Do you need to inject the `AgroalDataSource` or have the `ConnectionHolder`?

**Reply by nicklasweasel** [reply]
As I've understood it, DBRider looks for a ConnectionHolder to get a connection and I thought that one from the datasource would be a good fit. But apparently the factory method is never hit and goes deeper into DBRider until it comes to a place where the connection parameters are all empty strings.

This could of course be a DBRider thing but I've seen many examples constructed like that so I thought it might have something with conflicting lifecycles between DBRider and Quarkus i Junit5...

**Reply by edeandrea** [reply]
I don't know much about DBRider to be honest. I only looked at the docs/samples and found https://github.com/database-rider/database-rider/blob/master/rider-examples/quarkus-postgres-sample/src/test/java/com/github/quarkus/postgres/DbRiderIssue253It.java and noticed they weren't there. Granted the example is also a VERY old version of Quarkus.

**Comment by nicklasweasel** [comment]
The odd thing is that it if I make a plain java project, including

```
        <dependency>
            <groupId>com.github.database-rider</groupId>
            <artifactId>rider-junit5</artifactId>
            <version>1.42.0</version>
            <scope>test</scope>
        </dependency>
```

and do a

```
@DBRider
class DbTest {

    private final ConnectionHolder connectionHolder = () -> {
        System.out.println("Called connectionHolder");
        return null;
    };

    @Test
    void woik() {
        assertTrue(true);
    }
}
```

I see the output (and it bombs because of the null connection)

If I start with code-with-quarkus and do the same thing, the connectionHolder is never called (and the test bombs out with no connection). It's a bit odd that just being in a Quarkus project (no @QuarkusTest) should make DBRider skip the ConnectionHolder
